plugins/module_utils/network/ios/argspec/bgp_address_family/bgp_address_family.py

```diff
@@ -668,14 +668,25 @@ class Bgp_address_familyArgs(object):  # pylint: disable=R0903
                                     "type": "dict",
                                     "options": {
                                         "process_id": {"type": "int"},
+                                        "include_connected": {"type": "bool"},
                                         "match": {
                                             "type": "dict",
                                             "options": {
-                                                "external": {"type": "bool"},
                                                 "internal": {"type": "bool"},
-                                                "nssa_external": {"type": "bool"},
-                                                "type_1": {"type": "bool"},
-                                                "type_2": {"type": "bool"},
+                                                "external": {
+                                                    "type": "dict",
+                                                    "options": {
+                                                        "type_1": {"type": "bool"},
+                                                        "type_2": {"type": "bool"},
+                                                    },
+                                                },
+                                                "nssa_external": {
```

Comment:
@bentole we ideally deprecate the attributes which are not perfect by type and re-introduce a new set of attributes, to address it as a bugfix, or else they become a breaking change that requires a major release.

Reply:
You may refer to the PR https://github.com/ansible-collections/cisco.ios/pull/687/files for reference on how to deprecate, incase the tonne of things done in this PR confuse you let me know, I would push changes to this PR directly.
Regards

Reply:
Hi @KB-perByte ! I have made an attempt, emphasis on attempt, to address the issues you were requesting. Ideally I'd like to keep the `external` and `nssa_external` params just changing their type from `bool` to `dict`, but that's probably not possible until the old ones are removed?  Instead I introduced `externals` and `nssa_externals`. Also made an attempt to rewrite the changelog fragment. Hope I'm closer to compliance 😃